extensions/sdk/src/Rally.ts

```diff
@@ -99,9 +99,49 @@ export class Rally {
       connectFirestoreEmulator(this._db, 'localhost', 8080);
     }
 
+    this.storeAttributionCodes()
+      .catch((err) => console.error("Could not store attribution codes:", err));
+
     onAuthStateChanged(this._auth, user => this.authStateChangedCallback(user));
   }
 
+  /**
+   * Attempt to fetch the attribution codes from the store page URL for this extension.
+   */
+  private async storeAttributionCodes() {
+    const attribution = await this.getAttributionCodes();
+    if (!(Object.keys(attribution).length === 0 && attribution.constructor === Object)) {
+      console.debug("Attribution codes already stored");
+      return;
+    }
+
+    // Study IDs are in camelCase, store IDs are in hyphen-case.
+    const camelToHyphenCase = str => str.replace(/[A-Z]/g, letter => `-${letter.toLowerCase()}`);
+    const storeId = camelToHyphenCase(this._options.studyId);
+
+    let storeUrl;
+    if ("getBrowserInfo" in browser.runtime) {
+      const browserInfo = await browser.runtime.getBrowserInfo();
+      if (browserInfo.name === "firefox") {
+        storeUrl = `https://addons.mozilla.org/en-US/firefox/addon/${storeId}/*`
+      }
+    } else {
+      storeUrl = `https://chrome.google.com/webstore/detail/${storeId}/*`;
+    }
+
+    const tabs = await browser.tabs.query({ url: storeUrl });
+    if (!(tabs.length > 0)) {
+      throw new Error("No store URLs present in open tabs");
+    }
+    const url = new URL(tabs[0].url);
+
+    ["source", "medium", "campaign", "term", "content"].forEach((key) => {
+      attribution[key] = url.searchParams.get(`utm_${key}`);
+    });
+    browser.storage.local.set({ attribution });
+    console.debug("Attribution codes stored:", attribution);
+  }
```

Comment:
Should we extract/refactor a method that does this?

Reply:
Hm, sorry which part do you mean? Extracting the params from the URL?

Reply:
This method is `storeAttributionCodes()` so it does seem weird for it to be doing any sort of fetching and parsing of URLs I guess... there could be a `getCodesFromStore` maybe?